exec/java-exec/src/main/java/org/apache/drill/exec/service/ServiceEngine.java

```diff
@@ -56,22 +60,83 @@
   private final DrillConfig config;
   boolean useIP = false;
   private final boolean allowPortHunting;
+  private final BufferAllocator userAllocator;
+  private final BufferAllocator controlAllocator;
+  private final BufferAllocator dataAllocator;
+
 
   public ServiceEngine(ControlMessageHandler controlMessageHandler, UserWorker userWorker, BootStrapContext context,
       WorkEventBus workBus, WorkerBee bee, boolean allowPortHunting) throws DrillbitStartupException {
+    userAllocator = newAllocator(context, "rpc:user", "drill.exec.rpc.user.server.memory.reservation",
+        "drill.exec.rpc.user.server.memory.maximum");
+    controlAllocator = newAllocator(context, "rpc:bit-control",
+        "drill.exec.rpc.bit.server.memory.control.reservation", "drill.exec.rpc.bit.server.memory.control.maximum");
+    dataAllocator = newAllocator(context, "rpc:bit-control",
```

Comment:
Someone argued recently that it doesn't make sense to put everything there and I am in agreement. I've started adopting keeping strings where they are relevant.